src/index.ts

```diff
@@ -1,147 +1,172 @@
-   * insufficient data or concerning a user that no longer exists.
-   *
-   * A `5xx` status indicates a temporary internal error; this will be retried
-   * according to the step and function's retry policy (3 times, by default).
-   *
-   * @link https://www.inngest.com/docs/functions/function-input-and-output#response-format
-   * @link https://www.inngest.com/docs/functions/retries
-   */
-  status: number;
-
-  /**
-   * The output of the function - the `body` - can be any arbitrary
-   * JSON-compatible data. It is then usable by any future steps.
-   *
-   * @link https://www.inngest.com/docs/functions/function-input-and-output#response-format
-   */
-  body?: any;
-}
-
-/**
- * A single step within a function.
- */
-export type InngestStep<Context = any> = (
+export declare namespace Inngest {
   /**
-   * The context for this step, including the triggering event and any previous
-   * step output.
+   * An HTTP-like, standardised response format that allows Inngest to help
+   * orchestrate steps and retries.
    */
-  context: Context
-) => Promise<InngestResponse> | InngestResponse;
+  export interface Response {
+    /**
+     * A step response must contain an HTTP status code.
+     *
+     * A `2xx` response indicates success; this is not a failure and no retry is
+     * necessary.
+     *
+     * A `4xx` response indicates a bad request; this step will not be retried as
+     * it is deemed irrecoverable. Examples of this might be an event with
+     * insufficient data or concerning a user that no longer exists.
+     *
+     * A `5xx` status indicates a temporary internal error; this will be retried
+     * according to the step and function's retry policy (3 times, by default).
+     *
+     * @link https://www.inngest.com/docs/functions/function-input-and-output#response-format
+     * @link https://www.inngest.com/docs/functions/retries
+     */
+    status: number;
+
+    /**
+     * The output of the function - the `body` - can be any arbitrary
+     * JSON-compatible data. It is then usable by any future steps.
+     *
+     * @link https://www.inngest.com/docs/functions/function-input-and-output#response-format
+     */
+    body?: any;
+  }
 
-/**
- * The event payload structure for sending data to Inngest
- */
-export interface EventPayload {
-  /**
-   * A unique identifier for the event
-   */
-  name: string;
   /**
-   * Any data pertinent to the event
+   * A single step within a function.
    */
-  data: {
-    [key: string]: any;
-  };
+  export type Step<Context = any> = (
+    /**
+     * The context for this step, including the triggering event and any previous
+     * step output.
+     */
+    context: Context
+  ) => Promise<Response> | Response;
+
   /**
-   * Any user data associated with the event
-   * All fields ending in "_id" will be used to attribute the event to a particular user
+   * The event payload structure for sending data to Inngest
    */
-  user?: {
```

Comment:
I do really like the use of generics here, but our current generated types are for the entire Event Payload as to include data and user. That may be at odds with this, unless we perhaps change how the types are generated (or use `Pick` to extract the `data` and `user`).

Reply:
@djfarrelly This may be me trying to standardise something that's not standardised. 😄 We can shift this to just include the entire block instead? Whichever is easiest when porting the type generation to a CLI command.

Reply:
Yeah - I guess we use the entire payload right now, but we're still so early, we could def be subject to change some of this if you have better ideas!